lib/webpacker/compiler.rb

```diff
@@ -13,13 +13,17 @@ def initialize(webpacker)
 
   def compile
     if stale?
+      clobber
       record_compilation_timestamp
       run_webpack
-    else
-      logger.debug "No compiling needed as everything is fresh"
     end
   end
 
+  def clobber
+    config.public_output_path.rmtree if config.public_output_path.exist?
+    config.cache_path.rmtree if config.cache_path.exist?
+  end
```

Comment:
Sprockets uses 'clean' for "normal usage" cleanup -- discarding older versions of compiled assets, for example -- and 'clobber' for the nuclear "trash everything, even if reusing it "shouldn't" have any effect on the output".

Thus, clean is a thing one does when things are getting messy, and clobber is a thing one only really does when trying to debug a compilation problem.

At a quick glance, this might straddle those definitions, so I'll stop at providing that history/context, and won't venture a guess at which way this should go.